File: src/day10.rs
fn get_input() -> Vec<i64> {
    let input: Vec<&'static str> = include_str!("./inputs/day10.txt").lines().collect();
    input.iter().map(|num| num.parse::<i64>().unwrap()).collect()
}

fn joltage_differences() -> i64 {
    let mut joltage_ratings = get_input();
    let mut outlet: i64 = 0;
    let mut diff_1 = 0;
    let mut diff_3 = 1; // because of the highest rated adapter rule
    joltage_ratings.sort();

    for rating in joltage_ratings.iter() {
        let rating_value = *rating;
        if outlet + 1 == rating_value {
            diff_1 += 1;
        } else if outlet + 3 == rating_value {
            diff_3 += 1;
        }

        outlet = rating_value;
    }

    diff_1 * diff_3
}

pub fn get_solution_pt_1() -> i64 {
    joltage_differences()
}

File: deno/day10.ts
function parseInput(input: string) {
  return input.split("\n").map((num) => parseInt(num));
}

let joltageRatings: number[] = parseInput(await Deno.readTextFile('./inputs/day10.txt'));

function part1(joltageRatings: number[]) {
  let outlet = 0;
  let diff1 = 0;
  let diff3 = 1; // bc of the highest rated joltage rule

  joltageRatings.sort((a: number, b: number) => a - b);
  joltageRatings.forEach((rating: number) => {
    if (outlet + 1 === rating) {
      diff1 += 1;
    } else if (outlet + 3 === rating) {
      diff3 += 1;
    }

    outlet = rating;
  });
  return diff1;
}

console.log(part1(joltageRatings));
